lib/storage/test_storage.ex

```diff
@@ -1,35 +1,114 @@
 defmodule Uploadex.TestStorage do
   @moduledoc """
-  Storage for tests.
+  Storage to be used for testing. It holds the files in memory (using a `Agent`).
 
-  See files_test.exs for examples on how to use this.
+  ## Configuring
+
+  To configure your app to use the `Uploadex.TestStorage`, just add this storage to your `Uploadex` module,
+  depending on the current environment:
+
+      defmodule MyApp.Uploader do
+        use Uploadex
+
+        @impl true
+        def get_fields(%User{}), do: :files
+
+        @impl true
+        def default_opts(Uploadex.FileStorage), do: [...]
+        def default_opts(Uploadex.TestStorage), do: []
+
+        @impl true
+        def storage(%User{}, _field) do
+          if test_environment?() do
+            {Uploadex.TestStorage, []}
+          else
+            {Uploadex.FileStorage, [...]}
+          end
+        end
+
+        @impl true
+        def accepted_extensions(%User{}, _field), do: ~w(.jpg .png)
+
+        defp test_environment?() do
+          # This env must be set in your config files depending on the environment.
+          Application.fetch_env!(:my_app, :environment) == :test
+        end
+      end
+
+  ## Using
+
+  In your `ExUnit` tests, add a `setup` block starting the storage:
+
+      setup do
+        Uploadex.TestStorage.start_link()
+        :ok
+      end
+
+  Then, in the tests, you can use the `get_stored/1`, `get_deleted/1` and `get_opts`:
+
+      test "some test with files" do
+        assert ["document-1.pdf", "document-2.pdf"] == Uploadex.TestStorage.get_stored()
+        assert ["deleted-document.pdf"] == Uploadex.TestStorage.get_deleted()
+        assert [] == Uploadex.TestStorage.get_opts()
+      end
+
+  This module is just the base for testing, check `Uploadex.Testing` for a more convenient way to test uploads.
   """
 
   use Agent
 
   @behaviour Uploadex.Storage
 
-  def start_link(initial_value \\ %{stored: [], deleted: [], opts: []}) do
-    Agent.start_link(fn -> initial_value end, name: __MODULE__)
+  def start_link(initial_state \\ %{}) when is_map(initial_state) do
+    agent_name = initial_state |> Map.get(:opts, []) |> get_agent_name()
```

Comment:
what is the use case for accepting the name? Code code be simplified if always used the PID

Reply:
It's to make the TestStorage more flexible. For example, Bamboo has the `shared` option https://github.com/thoughtbot/bamboo/blob/main/lib/bamboo/test.ex#L99 that makes it work across processes. Ecto. has a similar thing. 

I didn't want to implement this in the lib, because I'm not exactly sure about the use cases, but at least I wanted to make the TestStorage flexible if any use case appears so the users of the lib can implement the way they wanted to and not get blocked by this.

Reply:
Hmm, okay.

Do you think it makes sense to implement a private `get` and `update` function that handles the logic to get the agent name? Would simplify the code a bit, but not super relevant